### PR TITLE
(DOCSP-21766): Added a note about the optional connectionStringSecretName field.

### PR DESCRIPTION
**Pull request by corryroot**
Hi @jwilliams-mongo, @mircea-cosbuc, and @irajdeep, I added a note about the optional `connectionStringSecretName` field.

[JIRA](https://jira.mongodb.org/browse/DOCSP-21766)

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: docs/deploy-configure.md [hunk 1]
```diff
@@ -38,6 +38,11 @@ To deploy your first replica set:
...
    | `<username>` | Username of the database user. | `my-user` |
 
+   **NOTE**: Alternatively, you can specify an optional 
+   `users[i].connectionStringSecretName` field to specify 
+   the name of the connection string secret that the 
+   Community Kubernetes Operator creates.
+   
    Update the variables in the following command, then run it to retrieve a user's connection strings to the replica set from the secret:
 
    **NOTE**: The following command requires [jq](https://stedolan.github.io/jq/) version 1.6 or higher.</br></br>
```
Comment (jwilliams-mongo) [inline docs/deploy-configure.md:48]: In the example on line 51, consider replacing the specific secret variables with a catch-all variable that also covers the note you added

**Review by jwilliams-mongo — APPROVED** [review]
@corryroot LGTM with one suggestion

File: docs/deploy-configure.md (hunk 1 above)
Comment (jwilliams-mongo) [reply docs/deploy-configure.md:48]: I think my comment from the enterprise PR might also apply here

**Review by mircea-cosbuc — APPROVED** [review]
LGTM

**Review by irajdeep — APPROVED** [review]
LGTM

**Pull request merged**